python/cudf/cudf/core/dataframe.py

```diff
@@ -3720,6 +3720,98 @@ def sort_values(
             keep_index=not ignore_index,
         )
 
+    def agg(self, aggs, axis=None):
+        """ 
+        Aggregate using one or more operations over the specified axis.
+
+        Parameters
+        ----------
+        aggs : str, list or dict
+            Function to use for aggregating data. Accepted combinations are:
+                string name (ex: "sum")
+                list of functions  (ex: ["sum", "min", "max"])
+                dict of axis labels specified operations per column
+                function name: must pass to DataFrame or DataFrame.apply
+
+        axis : {0 or ‘index’, 1 or ‘columns’}, default 0
+            If 0 or ‘index’: apply aggregation to each column
+            If 1 or ‘columns’: apply aggregation to each row.
+        
+        Returns 
+        -------
+        scalar : when Series.agg is called with single aggregation
+        Series : when DataFrame.agg is called with a single aggrgation
+        DataFrame : when DataFrame.agg is called with several aggregations
```

Comment:
I think in Sphinx one line of description refers to a separate parameter/return value right? Would recommend:
```suggestion
        Aggregation Result : scalar, ``Series`` or ``DataFrame``
                When Series.agg is called with single aggregation, a scalar is returned. 
                When DataFrame.agg is called with a single aggrgation, a ``Series`` is returned
                When DataFrame.agg is called with several aggregations, a ``DataFrame`` is returned
```